modules/mlflow/mlflow-fargate/stack.py

```diff
@@ -0,0 +1,206 @@
+from aws_cdk import aws_ec2 as ec2
+from aws_cdk import aws_ecr as ecr
+from aws_cdk import aws_ecs as ecs
+from aws_cdk import aws_ecs_patterns as ecs_patterns
+from aws_cdk import aws_efs as efs
+from aws_cdk import aws_iam as iam
+from aws_cdk import aws_s3 as s3
+from cdk_nag import AwsSolutionsChecks, NagPackSuppression, NagSuppressions
+from constructs import Construct, IConstruct
+
+
+class MlflowFargateStack(Stack):  # type: ignore
+    def __init__(
+        self,
+        scope: Construct,
+        id: str,
+        app_prefix: str,
+        vpc_id: str,
+        subnet_ids: List[str],
+        ecs_cluster_name: Optional[str],
+        service_name: Optional[str],
+        ecr_repo_name: str,
+        task_cpu_units: int,
+        task_memory_limit_mb: int,
+        autoscale_max_capacity: int,
+        artifacts_bucket_name: str,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(scope, id, **kwargs)
+
+        Tags.of(scope=cast(IConstruct, self)).add(key="Deployment", value=app_prefix[:64])
+
+        role = iam.Role(
+            self,
+            "TaskRole",
+            assumed_by=iam.ServicePrincipal(service="ecs-tasks.amazonaws.com"),
+            managed_policies=[
+                iam.ManagedPolicy.from_aws_managed_policy_name("AmazonECS_FullAccess"),
+            ],
+        )
+
+        # Grant artifacts bucket read-write permissions
+        model_bucket = s3.Bucket.from_bucket_name(self, "ArtifactsBucket", bucket_name=artifacts_bucket_name)
+        model_bucket.grant_read_write(role)
+
+        vpc = ec2.Vpc.from_lookup(self, "Vpc", vpc_id=vpc_id)
+        subnets = [ec2.Subnet.from_subnet_id(self, f"Sub{subnet_id}", subnet_id) for subnet_id in subnet_ids]
+
+        cluster = ecs.Cluster(
+            self,
+            "EcsCluster",
+            cluster_name=ecs_cluster_name,
+            vpc=vpc,
+            container_insights=True,
+        )
+        self.cluster = cluster
+
+        task_definition = ecs.FargateTaskDefinition(
+            self,
+            "MlflowTask",
+            task_role=role,
+            cpu=task_cpu_units,
+            memory_limit_mib=task_memory_limit_mb,
+        )
+
+        container = task_definition.add_container(
+            "ContainerDef",
+            image=ecs.ContainerImage.from_ecr_repository(
+                repository=ecr.Repository.from_repository_name(
+                    self,
+                    "ECRRepo",
+                    repository_name=ecr_repo_name,
+                ),
+            ),
+            environment={
+                "BUCKET": f"s3://{artifacts_bucket_name}",
+                "DUMMY": "DUMMY",
+            },
+            logging=ecs.LogDriver.aws_logs(stream_prefix="mlflow"),
+        )
+        port_mapping = ecs.PortMapping(container_port=5000, host_port=5000, protocol=ecs.Protocol.TCP)
+        container.add_port_mappings(port_mapping)
+
+        # Add EFS
+        fs = efs.FileSystem(
```

Comment:
Is it a good idea to externalize EFS from this module and consume from IDF kind of a generic repo?

Reply:
Yeah I think it s a good idea, but only after we have a second use case for it, to make sure we actually have a generalized version of it